CONTRIBUTING.md

```diff
@@ -1,47 +1,297 @@
-# How to contribute
+# Contribution Guide #
+TiDB is a community driven open source project. All contributor are very welcome. The process of contributing to the TiDB project
+may be different than many other projects you get involved in. This document outlines some conventions about development workflow, commit message formatting, contact points and other resources to make it easier to get your contribution accepted. This document is the canonical source of truth for things like supported toolchain versions for building and testing TiDB.
 
-This document outlines some of the conventions on development workflow, commit message formatting, contact points and other
-resources to make it easier to get your contribution accepted.
+Please submit an [issue] on github if you 
+* find a requirement that this doc does not capture,
+* find other docs with references to requirements that
+  are not simply links to this doc.
 
-## Getting started
+## Pre submit flight checks
 
-- Fork the repository on GitHub.
-- Read the README.md for build instructions.
-- Play with the project, submit bugs, submit patches!
+Make sure you decide whether your issue and/or pull request is improving TiDB architecture or whether its simply fixing a bug.
 
-## Contribution flow
+Make sure there are no typos, if you need a diagram, add it.  Make sure you SEPARATE the description of the problem
+(i.e. Y is a critical component that is too slow for an SLA that we care about) from the solution (i.e. make X faster).
 
-This is a rough outline of what a contributor's workflow looks like:
+In order to save reviewers time, each issue should be file with card and should be sanity-checkable in under 5 minutes(even 
```

Comment:
it should be "be filed with card". It means issue should come with a template. sanity-checkable means it can verify very quickly(i.e. how to reproduce a bug)